sdk/maps/azure-maps-geolocation/src/test/java/com/azure/maps/geolocation/GeolocationClientTestBase.java

```diff
@@ -3,95 +3,116 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class GeolocationClientTestBase extends TestProxyTestBase {
+    private static final String SDK_NAME = "client_name";
+    private static final String SDK_VERSION = "client_version";
+
     static final String FAKE_API_KEY = "fakeKeyPlaceholder";
 
-    private final String endpoint = Configuration.getGlobalConfiguration().get("API-LEARN_ENDPOINT");
-    Duration durationTestMode;
     static InterceptorManager interceptorManagerTestBase;
 
+    Duration durationTestMode;
+
     @Override
     protected void beforeTest() {
         if (interceptorManager.isPlaybackMode()) {
             durationTestMode = Duration.ofMillis(1);
         } else {
             durationTestMode = TestUtils.DEFAULT_POLL_INTERVAL;
         }
+
         interceptorManagerTestBase = interceptorManager;
     }
 
-    GeolocationClientBuilder getGeoLocationAsyncClientBuilder(HttpClient httpClient, GeolocationServiceVersion serviceVersion) {
+    GeolocationClientBuilder getGeoLocationAsyncClientBuilder(HttpClient httpClient,
+                                                              GeolocationServiceVersion serviceVersion) {
         GeolocationClientBuilder builder = new GeolocationClientBuilder()
+            .pipeline(getHttpPipeline(httpClient))
             .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
             .serviceVersion(serviceVersion);
-        String endpoint = getEndpoint();
-        if (getEndpoint() != null) {
-            builder.endpoint(endpoint);
-        }
-        if (getTestMode() == TestMode.RECORD) {
-            builder.addPolicy(interceptorManager.getRecordPolicy());
-        }
-        if (getTestMode() == TestMode.PLAYBACK) {
-            builder.credential(new AzureKeyCredential(FAKE_API_KEY)).httpClient(interceptorManager.getPlaybackClient());
-        } else {
-            builder.credential((new AzureKeyCredential(
-                Configuration.getGlobalConfiguration().get("SUBSCRIPTION_KEY"))));
+
+        if (interceptorManager.isPlaybackMode()){
+            builder.endpoint("https://localhost:8080");
         }
+
         return builder;
     }
 
     HttpPipeline getHttpPipeline(HttpClient httpClient) {
-        TokenCredential credential = null;
+        httpClient = interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient;
 
-        if (!interceptorManager.isPlaybackMode()) {
-            credential = new EnvironmentCredentialBuilder().httpClient(httpClient).build();
+        if (interceptorManager.isRecordMode()) {
+            interceptorManager.addSanitizers(
+                Collections.singletonList(
+                    new TestProxySanitizer("subscription-key", ".+", "REDACTED", TestProxySanitizerType.HEADER)));
```

Comment:
Can you add this key here https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestProxyUtils.java#L65

So that its covered for other sdks too, that might be using it.

Reply:
Created this PR: https://github.com/Azure/azure-sdk-for-java/pull/35872